Jogo da Cobrinha

YasminDeodato opened the pull request:
Versão 1 - Jogo da Cobrinha
Disponível em: http://www.codeskulptor.org/#user46_xuO71DPRdk_14.py

YasminDeodato committed:
Jogo da Cobrinha

Versão 1 - Jogo da Cobrinha
Disponível em: http://www.codeskulptor.org/#user46_xuO71DPRdk_14.py

flaviobarros commented:
Muito bem Yasmin! Você está no caminho certo. Acho que agora eu tenho três sugestões:

1) desenhar os "gomos" da cobrinha com exatamente um raio para trás de forma que a cobrinha apareça mais completa.

2) quando você sorteia a posição das comidas o ideal era que ela estivesse alinhada com o caminho da cobra. 

3) falta implementar a lógica para os outros gomos seguirem o caminho da cobrinha.

É isso aí.

flaviobarros commented on Yasmin/jogo_cobrinha.py:
```diff
@@ -0,0 +1,197 @@
+            # move os gomos em ordem reversa
+            for index in range(len(corpo) - 1, 0, -1):                        
+                        x = corpo[index - 1][0] - BOLA_RAIO * vel[0] 
+                        y = corpo[index - 1][1] - BOLA_RAIO * vel[1]  
```
Aqui se você subtrair 2*BOLA_RAIO você vai desenhar a cobra certinho, uma bolinha atrás da outra:

x = corpo[index - 1][0] - 2*BOLA_RAIO * vel[0] 
y = corpo[index - 1][1] - 2*BOLA_RAIO * vel[1]

flaviobarros commented on Yasmin/jogo_cobrinha.py:
```diff
@@ -0,0 +1,197 @@
+                        
+            # move o primeiro segmento do corpo p/ posicao da cabeca 
+            if len(corpo) > 0:
+                corpo[0] = [cabeca[0] - BOLA_RAIO * vel[0], cabeca[1] -  BOLA_RAIO * vel[1]]
```
Aqui também:

corpo[0] = [cabeca[0] - 2*BOLA_RAIO * vel[0], cabeca[1] -  2*BOLA_RAIO * vel[1]]

flaviobarros commented on Yasmin/jogo_cobrinha.py:
```diff
@@ -0,0 +1,197 @@
+    # desenha o corpo da cobra
+    if corpo:
+        for gomo in corpo:
+            canvas.draw_circle(gomo, 10, 1, "White", "White")
```
Ao invés de usar um RAIO de 10 utilize o mesmo raio da bola:

canvas.draw_circle(gomo, BOLA_RAIO, 1, "White", "White")

flaviobarros commented on Yasmin/jogo_cobrinha.py:
```diff
@@ -0,0 +1,197 @@
+    # calcula a distancia entre a cabeça da cobra e a comida
+    distancia = math.sqrt(((cabeca[0]-comida_pos[0])**2)+((cabeca[1]-comida_pos[1])**2))   
+    
+    if distancia <= 5:
```
aqui eu utilizaria o critério de que se as duas bolas tangenciarem você já comeria:

então o teste seria:

if distancia <= BOLA_RAIO + COMIDA_RAIO:

flaviobarros reviewed (changes requested):
Dá uma olhada nas minhas sugestões que aí só faltará resolver os outros dois problemas.

YasminDeodato committed:
Atualização - Jogo da Cobrinha

Versão 1.1 - Jogo da Cobrinha
Disponível em: http://www.codeskulptor.org/#user46_xTrdhFy2r3_1.py

Alterações:
* Os "gomos" da cobrinha estão desenhados exatamente um raio para trás de forma que a cobrinha apareça mais completa.
* Uma cobrinha come a comida quando a bola da cabeça e a da comida se tangenciam.

YasminDeodato committed:
Atualização Jogo da Cobrinha

YasminDeodato committed:
Atualização - Formatador de Tempo
